pyseir/cli.py

```diff
@@ -74,21 +62,97 @@ def _states_region_list(state: Optional[str], default: List[str]) -> List[pipeli
         return [pipeline.Region.from_state(s) for s in default]
 
 
-def _state_only_pipeline(
-    region: pipeline.Region, run_mode=DEFAULT_RUN_MODE, output_interval_days=1, output_dir=None,
-) -> model_fitter.ModelFitter:
-    assert region.is_state()
+@dataclass
+class StatePipeline:
+    """Runs the pipeline for one state and stores the output."""
+
+    region: pipeline.Region
+    infer_df: pd.DataFrame
+    fitter: model_fitter.ModelFitter
+
+    @staticmethod
+    def run(region: pipeline.Region, run_mode: pyseir.utils.RunMode) -> "StatePipeline":
+        assert region.is_state()
+        infer_df = infer_rt.run_rt(infer_rt.RegionalInput.from_region(region))
+        fitter_input = model_fitter.RegionalInput.from_state_region(region)
+        fitter = model_fitter.ModelFitter.run_for_region(fitter_input)
+        ensembles_input = ensemble_runner.RegionalInput.for_state(fitter)
+        ensemble_runner.run_region(ensembles_input, ensemble_kwargs={"run_mode": run_mode})
+        return StatePipeline(region=region, infer_df=infer_df, fitter=fitter)
+
+
+@dataclass
+class SubStateRegionPipelineInput:
+    region: pipeline.Region
+    run_fitter: bool
+    state_fitter: model_fitter.ModelFitter
+    run_mode: pyseir.utils.RunMode
+
+    @staticmethod
+    def build_all(
+        fips: Optional[str],
```

Comment:
This feels like it should be a keyword arg?

Reply:
Done